src/chat.rs

```diff
@@ -926,6 +934,46 @@ impl Chat {
     }
 }
 
+#[derive(Debug, PartialEq, Clone)]
+pub enum ArchiveState {
+    /// Neither archived or pinned
+    Normal = 0,
+    Archived = 1,
+    /// Pinned (formaly known as sticky)
+    Pinned = 2,
+}
+
+impl rusqlite::types::ToSql for ArchiveState {
+    fn to_sql(&self) -> rusqlite::Result<rusqlite::types::ToSqlOutput> {
+        let duration = match &self {
+            ArchiveState::Normal => 0,
+            ArchiveState::Archived => 1,
+            ArchiveState::Pinned => 2,
+        };
+        let val = rusqlite::types::Value::Integer(duration as i64);
+        let out = rusqlite::types::ToSqlOutput::Owned(val);
+        Ok(out)
+    }
+}
+
+impl rusqlite::types::FromSql for ArchiveState {
+    fn column_result(value: rusqlite::types::ValueRef) -> rusqlite::types::FromSqlResult<Self> {
+        i64::column_result(value).and_then(|val| {
+            Ok({
+                match val {
+                    2 => ArchiveState::Pinned,
+                    1 => ArchiveState::Archived,
+                    0 => ArchiveState::Normal,
+                    _ => {
+                        println!("unknown archived state, falling back to normal state (was this db opened with a newer deltachat version?)");
```

Comment:
No printlns please.  try and handle this similar to how mute was done, e.g. an OutOfRange error or so.